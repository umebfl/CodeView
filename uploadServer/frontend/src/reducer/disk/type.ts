export interface DiskType {
    id: number
    diskId: string
    owner: DiskOwnerType
    inventoryStatus: DiskInventoryStatusType
    onServer: boolean
    comment: string
    unregistered?: boolean
}

export interface DiskUploadRecordType {
    uploadStartTime: string
    uploadEndTime: string
    vehicleId: string
    xrayUris: string[]
    uploadStatus: DiskUploadStatus
}

export enum DiskUploadStatus {
    NOT_UPLOAD = 0,
    UPLOADING = 1,
    SUCCEED = 2,
    FAILED = 3,
}

export interface DiskResponseType {
    disk_sn: string
    owner: DiskOwnerType
    status: DiskInventoryStatusType
    on_server: boolean
    comment: string
}

export enum DiskOwnerType {
    UNKNOWN = 0,
    PINGSHAN = 1,
    NANSHAN = 2,
}

export enum DiskInventoryStatusType {
    NORMAL = 0,
    LOST = 1,
    DAMAGED = 2,
}

export interface diskState {
    data: DiskType[]
    uploadRecords: Record<string, DiskUploadRecordType[]>
}
