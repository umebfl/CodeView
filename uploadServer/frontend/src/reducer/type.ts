import { RematchDispatch, RematchRootState } from '@rematch/core'

import { RootModel } from '.'
import { store } from './store'

export type Store = typeof store
export type Dispatch = RematchDispatch<RootModel>
export type RootState = RematchRootState<RootModel>
