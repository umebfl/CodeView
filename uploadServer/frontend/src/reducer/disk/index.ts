import {
    addIndex,
    find,
    findIndex,
    keys,
    map,
    propEq,
    reduce,
    take,
    update,
} from 'ramda'
import { createModel } from '@rematch/core'

import request from 'src/util/request'

import { RootModel } from '..'
import { DiskType, diskState, DiskResponseType } from 'src/reducer/disk/type'
import { DEFAULT_DATA_SOURCE } from 'src/reducer/userConfig'

const MAX_COMMENT_LEN = 120

const initState: diskState = {
    data: [],
    uploadRecords: {},
}

export const disk = createModel<RootModel>()({
    state: initState,
    reducers: {
        setData: (state, payload: Partial<diskState>) => {
            return {
                ...state,
                ...payload,
            }
        },
    },
    effects: dispatch => ({
        async initData(_, rootState) {
            const { dataSource } = rootState.userConfig

            const ds = dataSource || DEFAULT_DATA_SOURCE

            const data: {
                disks_info: DiskResponseType[]
            } = await request({
                url: `/disk_management/get_disks_info?region=${ds}`,
                rootState,
                dispatch,
            })

            const { disks_info } = data

            if (disks_info) {
                const transData: DiskType[] = addIndex<DiskResponseType>(map)(
                    (disk: DiskResponseType, idx: number): DiskType => ({
                        id: idx,
                        diskId: disk.disk_sn,
                        owner: disk.owner,
                        inventoryStatus: disk.status,
                        onServer: disk.on_server,
                        comment: disk.comment,
                    })
                )(disks_info)

                dispatch.disk.setData({
                    data: transData,
                })
            }
        },

        async getUploadRecords(payload: string, rootState) {
            try {
                const data = await request({
                    url: `/disk_management/get_upload_records?disk_sn=${payload}`,
                    rootState,
                    dispatch,
                })
                const { disk } = rootState

                if (data.upload_records) {
                    dispatch.disk.setData({
                        uploadRecords: {
                            ...disk.uploadRecords,
                            [payload]: addIndex(map)(
                                (record: any, idx: number) => ({
                                    id: idx,
                                    seq: idx,
                                    uploadServerId: record.upload_server_id,
                                    uploadStartTime: record.upload_start_time,
                                    uploadEndTime: record.upload_end_time,
                                    diskPlugTime: record.disk_plug_time,
                                    vehicleId: record.vehicle_id,
                                    xrayUris: record.xray_uris,
                                    uploadStatus: record.upload_status,
                                })
                            )(data.upload_records),
                        },
                    })
                }
            } catch (error) {
                console.error(error)
            }
        },

        async stopUpload(
            payload: {
                serverId: string
                diskId: string
            },
            rootState
        ) {
            try {
                const rv = await request({
                    url: `/data_center/set_disk_finished_manually?upload_server_id=${payload.serverId}&disk_sn=${payload.diskId}`,
                    rootState,
                    dispatch,
                })

                return rv
            } catch (error) {
                console.error(error)
            }
        },

        async changeDiskInfo(
            payload: {
                newRow: DiskType
                params: DiskResponseType
            },
            rootState
        ) {
            const { newRow, params } = payload

            const fixParams = {
                ...params,
                comment: take(MAX_COMMENT_LEN, params.comment || ''),
            }

            const paramsKeys = keys(fixParams)
            const paramStr = reduce(
                (val: string, key: keyof DiskResponseType) => {
                    return `${val}${key}=${fixParams[key]}&`
                },
                ''
            )(paramsKeys)

            try {
                const rv = await request({
                    url: `/disk_management/upsert_disk_info?${paramStr}`,
                    rootState,
                    dispatch,
                })

                if (rv?.code === 0) {
                    const idx = findIndex(
                        propEq('diskId', newRow.diskId),
                        rootState.disk.data
                    )

                    const fixData = update(idx, newRow, rootState.disk.data)

                    dispatch.disk.setData({
                        data: fixData,
                    })

                    return true
                }

                return false
            } catch (error) {
                console.error(error)
            }
        },
    }),
})
