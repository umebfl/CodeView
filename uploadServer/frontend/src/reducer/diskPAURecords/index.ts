import { createModel } from '@rematch/core'

import request from 'src/util/request'

import { RootModel } from '..'
import {
    diskPAURecordsType,
    diskPAURecordsState,
} from 'src/reducer/diskPAURecords/type'

const initState: diskPAURecordsState = {
    data: [],
}

export const diskPAURecords = createModel<RootModel>()({
    state: initState,
    reducers: {
        setData: (state, payload: diskPAURecordsType[]) => {
            return {
                ...state,
                data: payload,
            }
        },
    },
    effects: dispatch => ({
        async initData(_, rootState) {
            const data = await request({
                url: '/data_center/get_diskPAURecords_list',
                rootState,
                dispatch,
            })

            if (data?.diskPAURecords) {
                dispatch.diskPAURecords.setData(data.diskPAURecords)
            }
        },
    }),
})
