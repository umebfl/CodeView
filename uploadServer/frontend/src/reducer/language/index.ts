import { createModel } from '@rematch/core'

import { RootModel } from '..'
import { langSet, languagePropsType } from 'src/reducer/language/type'

const initState: languagePropsType = {
    lang: langSet.zh,
}

export const language = createModel<RootModel>()({
    state: initState,
    reducers: {
        setLang: (state, payload: langSet) => ({
            ...state,
            lang: payload,
        }),
        switchLang: state => ({
            ...state,
            lang: state.lang === langSet.en ? langSet.zh : langSet.en,
        }),
    },
})
