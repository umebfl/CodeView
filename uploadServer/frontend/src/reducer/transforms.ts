import { createTransform } from 'redux-persist'

import { DataSourceEnum } from 'src/reducer/userConfig/type'

const SetTransform = createTransform(
    // transform state on its way to being serialized and persisted.
    (inboundState: any, key) => {
        // old persisted state
        if (
            inboundState.dataSource &&
            (inboundState.dataSource === 'guangzhou' ||
                inboundState.dataSource === 'shenzhen')
        ) {
            return {
                ...inboundState,
                dataSource:
                    inboundState.dataSource === 'guangzhou'
                        ? DataSourceEnum.guangzhou
                        : DataSourceEnum.shenzhen,
            }
        }

        return inboundState
    },

    // transform state being rehydrated
    (outboundState: any, key) => {
        // old persisted state
        if (
            outboundState.dataSource &&
            (outboundState.dataSource == 'guangzhou' ||
                outboundState.dataSource == 'shenzhen')
        ) {
            return {
                ...outboundState,
                dataSource:
                    outboundState.dataSource === 'guangzhou'
                        ? DataSourceEnum.guangzhou
                        : DataSourceEnum.shenzhen,
            }
        }

        // convert mySet back to a Set.
        return outboundState
    },
    // define which reducers this transform gets called for.
    // { whitelist: ['someReducer'] }
    { whitelist: ['userConfig', 'language'] }
)

export default SetTransform
