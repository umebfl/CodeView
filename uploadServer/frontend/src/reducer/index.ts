/** No need unit test */
import { Models } from '@rematch/core'

import { uploadServer } from 'src/reducer/uploadServer'
import { globalLoading } from 'src/reducer/globalLoading'
import { snackbar } from 'src/reducer/snackbar'
import { userConfig } from 'src/reducer/userConfig'
import { language } from 'src/reducer/language'

export interface RootModel extends Models<RootModel> {
    uploadServer: typeof uploadServer
    globalLoading: typeof globalLoading
    snackbar: typeof snackbar
    userConfig: typeof userConfig
    language: typeof language
}

export const models: RootModel = {
    uploadServer,
    globalLoading,
    snackbar,
    userConfig,
    language,
}
