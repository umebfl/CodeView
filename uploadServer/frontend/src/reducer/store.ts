/** No need unit test */
import { init } from '@rematch/core'
import createRematchPersist from '@rematch/persist'
import AsyncStorage from '@react-native-community/async-storage'

import { models, RootModel } from 'src/reducer'

const persistPlugin = createRematchPersist<RootModel, any>({
    key: 'autox',
    storage: AsyncStorage,
    whitelist: ['userConfig', 'language'],
    throttle: 1000,
    debug: false,
    version: 6,
})

export const getInitStore = () =>
    init({
        models,
        plugins: [persistPlugin],
    })

export const store = getInitStore()
