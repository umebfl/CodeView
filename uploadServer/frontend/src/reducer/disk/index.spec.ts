import { init, createModel, Models } from '@rematch/core'
import { dissoc } from 'ramda'

import { rest } from 'msw'
import { setupServer } from 'msw/node'

import { disk } from 'src/reducer/disk'

const mockData = {
    data: [
        {
            comment: 'TEST_DISK_SN',
            diskId: '',
            id: 1,
            inventoryStatus: 0,
            onServer: true,
            owner: 2,
        },
    ],
    uploadRecords: {
        TEST_DISK_SN: {
            id: '1',
            seq: 0,
            uploadStartTime: '2022-01-02 11:22:12',
            uploadEndTime: '2022-02-02 11:22:12',
            vehicleId: 'byd-02',
            xrayUris: ['byd-2-records'],
            uploadStatus: 0,
        },
    },
}

describe('Reducer - disk', () => {
    it('should set data', () => {
        const store = init({
            models: { disk } as any,
        })

        const { dispatch } = store

        dispatch.disk.setData(mockData)

        const data = store.getState().disk
        expect(data).toEqual(mockData)
    })
})
