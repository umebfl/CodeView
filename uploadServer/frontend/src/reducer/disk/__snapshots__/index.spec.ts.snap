// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer - disk should get Upload Records data 1`] = `
Object {
  "data": Array [
    Object {
      "comment": "TEST_DISK_SN",
      "diskId": "",
      "id": 1,
      "inventoryStatus": 0,
      "onServer": true,
      "owner": 2,
    },
  ],
  "uploadRecords": Object {
    "S6CKNE0R920379": Array [
      Object {
        "id": 0,
        "seq": 0,
        "uploadEndTime": "1971-01-01 00:00:00",
        "uploadServerId": "uploadServerId-1",
        "uploadStartTime": "1971-01-01 00:00:00",
        "uploadStatus": 0,
        "vehicleId": "pacifica-cn-53",
        "xrayUris": Array [
          "pacifica-cn-53/20220419/2022-04-19-11-31-19",
          "pacifica-cn-53/20220419/2022-04-19-09-19-44",
          "pacifica-cn-53/20220419/2022-04-19-09-51-39",
        ],
      },
    ],
  },
}
`;

exports[`Reducer - disk should init data 1`] = `
Object {
  "data": Array [
    Object {
      "comment": "test_disk_info",
      "diskId": "UNKNOWN_DISK_ID_0000:60:00.0",
      "id": 0,
      "inventoryStatus": 0,
      "onServer": true,
      "owner": 1,
    },
    Object {
      "comment": "test_disk_info",
      "diskId": "A0765C71",
      "id": 1,
      "inventoryStatus": 1,
      "onServer": true,
      "owner": 2,
    },
  ],
  "uploadRecords": Object {},
}
`;

exports[`Reducer - disk should set data 1`] = `
Object {
  "data": Array [
    Object {
      "comment": "TEST_DISK_SN",
      "diskId": "",
      "id": 1,
      "inventoryStatus": 0,
      "onServer": true,
      "owner": 2,
    },
  ],
  "uploadRecords": Object {
    "TEST_DISK_SN": Object {
      "id": "1",
      "seq": 0,
      "uploadEndTime": "2022-02-02 11:22:12",
      "uploadServerId": "uploadServerId-1",
      "uploadStartTime": "2022-01-02 11:22:12",
      "uploadStatus": 0,
      "vehicleId": "byd-02",
      "xrayUris": Array [
        "byd-2-records",
      ],
    },
  },
}
`;
